src/aws-util/src/s3_uploader.rs

```diff
@@ -300,6 +300,7 @@ impl S3MultiPartUploader {
             return Err(S3MultiPartUploadError::ExceedsMaxPartNumber);
         }
 
+        // TODO: Spawn this on a new task to make it parallel.
```

Comment:
parallel? or just not doing io on timely workers?

Reply:
it's already not on a timely worker, but we can spawn this in a new task to allow the existing task to continue doing work while this part uploads in the background. There is no need we need to wait for this part to upload here